packages/schemas/src/foundations/jsonb-types/hooks.ts

```diff
@@ -29,3 +66,19 @@ export const hookConfigGuard = z.object({
 });
 
 export type HookConfig = z.infer<typeof hookConfigGuard>;
+
+/**
+ * Management API hooks registration.
+ *
+ * Pre register the hooks that will be triggered by the management API.
+ * Used by the managementApiHooks middleware.
+ * Auto register the following hooks when the management API is called.
+ */
```

Comment:
- this comment block will be applied to the `ApiMethod` type, which looks unexpected
- it looks not easy to follow the description. would you mind rephrasing it?